script.js

```diff
@@ -225,4 +225,40 @@ form.addEventListener('submit', (e) => {
     email.style.borderRadius = '3px';
     error.style.backgroundColor = 'red';
   }
-});
\ No newline at end of file
+});
+
+// Preserve Data on the Browser Section
+
+function getFormData() {
+  const formData = {
+    name: document.getElementById('name').value,
+    email: document.getElementById('mail').value,
+    message: document.getElementById('msg').value,
+  };
+
+  localStorage.setItem('formData', JSON.stringify(formData));
+}
+
+document.getElementById('name').addEventListener('change', () => {
+  getFormData();
+});
+document.getElementById('mail').addEventListener('change', () => {
+  getFormData();
+});
+document.getElementById('msg').addEventListener('change', () => {
+  getFormData();
+});
+
+function loadLocalStorageData() {
+  if (localStorage.getItem('formData') !== null) {
+    const retrieveFormData = localStorage.getItem('formData');
+    const retrieveJsonData = JSON.parse(retrieveFormData);
+    document.getElementById('name').value = retrieveJsonData.name;
+    document.getElementById('mail').value = retrieveJsonData.email;
+    document.getElementById('msg').value = retrieveJsonData.message;
+  }
+}
```

Comment:
- [ ] the email from doesn't preserve the data  in the local storage and also when I update the other form doesn't preserve the new data that I typed please try to fix that

Reply:
@danigalindo10 from my side doesn't work but it seems that it's working on your local machine so then just send for another review the reviewer will approve it

Reply:
Please can you ensure me that? This would be my last request to review.